core/tx_pool.go

```diff
@@ -1057,10 +1381,6 @@ func (pool *TxPool) promoteExecutables(accounts []common.Address) {
 		for _, tx := range list.Forward(nonce) {
 			hash := tx.Hash()
 			logger.Warn().Str("hash", hash.Hex()).Msg("Removed old queued transaction")
-			if pool.chain.CurrentBlock().Transaction(hash) == nil {
```

Comment:
why are these checks removed?

Reply:
Error report not needed here since error would be caught when tx is added to tx-pool in the first place (from the `validateTx` fn). Also looking up a tx is a linear time op in the number of txs in the block.